cats_tutorial.ipynb

```diff
@@ -0,0 +1,491 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Simulating a smart thermostat using CATS: a Contextual Bandit algorithm with a continuous action space\n",
+    "\n",
+    "In this tutorial we will simulate the scenario of personalizing a thermostat for a household with two rooms, using Contextual Bandits in a continuous action space. The goal is to maximize user satisfaction with the thermostat quantified by measuring thermostat accuracy or reward (TR). The thermostat proposes a temperature and the user will either accept the temperature or adjust it to fit their needs.\n",
+    "\n",
+    "Let's recall that in a CB setting, a data point has four components,\n",
+    "\n",
+    "Context\n",
+    "Action\n",
+    "Probability of choosing action\n",
+    "Reward/cost for chosen action\n",
+    "\n",
+    "In our simulator, we will need to generate a context, get an action/decision for the given context and also simulate generating a reward.\n",
+    "\n",
+    "In our simulator, our goal is to maximize reward (TR) or minimize loss (-TR)\n",
+    "\n",
+    "The thermostat tracks two rooms: 'Living Room' and 'Bedroom'\n",
+    "Each room will need temperature adjustment either in the morning or in the afternoon\n",
+    "The context is therefore (room, time_of_day)\n",
+    "\n",
+    "In a continuous range, we can't specify actions since there are infinite actions we can take across the continuous range. We do however provide the minimum value and the maximum value of the range. Here we will range between 0 degrees celsius and 32 degrees celsius which gives us a continuous range of 32 degrees.\n",
```

Comment:
I would add to the last sentence: "Here we will range between 0 degrees Celsius and 32 degrees Celsius using 1 degree increments which gives us a continuous range of 33 degrees.